system/helpers/numberUtils.cfm

```diff
@@ -0,0 +1,29 @@
+<cffunction name="numberThousandDecimalFormat" access="public" returntype="string" output="false">
+	<cfargument name="number" type="numeric" required="true" />
+	<cfargument name="decimals" type="numeric" required="false" default="0" />
+
+	<cfscript>
+		var decimalPlaceholder = "";
+
+		if( round( arguments.number ) != arguments.number || arguments.decimals > 0 ) {
+			var decimalsCount = 0;
```

Comment:
I think this could be simpler:

```cfc
var decimalPlaces = Len( ListRest( arguments.number, "." ) );
var numberFormat = decimalPlaces ? ",.#repeatString( "_", decimalPlaces )#" : ",";

return LsNumberFormat( arguments.number, numberFormat );
```

Reply:
Updated for simpler code.